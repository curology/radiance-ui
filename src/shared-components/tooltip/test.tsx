import React from 'react';
import { render } from 'src/tests/testingLibraryHelpers';

import { Tooltip } from './index';

describe('<Tooltip />', () => {
  describe('UI snapshot', () => {
    it('renders content and children', () => {
      const { container } = render(
        <Tooltip
          position="bottom"
          content="Tooltip Content"
        >
          Tooltip Trigger
        </Tooltip>,
      );

      expect(container.firstElementChild).toMatchSnapshot();
    });
  });
});
