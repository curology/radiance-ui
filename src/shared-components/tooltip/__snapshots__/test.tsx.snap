// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> UI snapshot renders content and children 1`] = `
.emotion-0 {
  position: relative;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
}

.emotion-2 {
  cursor: pointer;
}

.emotion-4 {
  max-width: none;
  top: 120%;
  background: #332E54;
  box-shadow: 0px 8px 24px rgba(51, 46, 84, 0.05);
  border-radius: 8px;
  color: #FFFFFF;
  min-width: 100px;
  opacity: 0;
  padding: 1rem;
  cursor: none;
  pointer-events: none;
  position: absolute;
  -webkit-transform: translateY(-8px);
  -moz-transform: translateY(-8px);
  -ms-transform: translateY(-8px);
  transform: translateY(-8px);
  transition-delay: 30ms;
  transition-timing-function: ease-in-out;
  -webkit-transition: 350ms;
  transition: 350ms;
  z-index: 5;
  text-align: left;
  font-size: 0.875rem;
  display: block;
}

.emotion-6 {
  z-index: 5;
  position: relative;
}

.emotion-8 {
  position: absolute;
  z-index: 4;
  top: -8px;
  -webkit-transform: rotate(180deg);
  -moz-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
  left: 50%;
  margin-left: -10px;
}

.emotion-10 {
  display: block;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  color: #332E54;
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  transition: color 350ms,transform 350ms;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <div
      class="emotion-2 emotion-3"
    >
      Tooltip Trigger
    </div>
    <div
      class="emotion-4 emotion-5"
    >
      <div
        class="emotion-6 emotion-7"
      >
        Tooltip Content
      </div>
      <div
        class="emotion-8 emotion-9"
      >
        <svg
          class="emotion-10"
          fill="#332E54"
          height="16"
          viewBox="0 0 46.29 32.92"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M27.07 30.75a4.63 4.63 0 01-7.85 0L0 0h46.27z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
